File: guides/micronaut-flyway/java/src/main/java/example/micronaut/Person.java
/*
 * Copyright 2017-2024 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut;
/*
//tag::package[]
package example.micronaut;
//end::package[]
*/
//tag::clazz1[]
import io.micronaut.core.annotation.NonNull;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Version;

import jakarta.validation.constraints.NotBlank;

@MappedEntity // <1>
public class Person {

    @Id // <2>
    @GeneratedValue // <3>
    private Long id;

    @Version // <4>
    private Long version;

    @NonNull
    @NotBlank
    private final String name;
    //end::clazz1[]

    //tag::nullableage[]
    @Nullable
    private final Integer age;

    public Person(@NonNull String name,
                  @Nullable Integer age) {
        this.name = name;
        this.age = age;
    }

    @Nullable
    public Integer getAge() {
        return age;
    }

    //end::nullableage[]

    /*
//tag::ageint[]
    private final int age;

    public Person(@NonNull String name, int age) {
        this.name = name;
        this.age = age;
    }

    public int getAge() {
        return age;
    }
//end::ageint[]

     */
//tag::clazz2[]

    @NonNull
    public String getName() {
        return name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getVersion() {
        return version;
    }

    public void setVersion(Long version) {
        this.version = version;
    }
}
//end::clazz2[]

File: guides/micronaut-liquibase/kotlin/src/main/kotlin/example/micronaut/Person.kt
/*
 * Copyright 2017-2023 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
//tag::clazz1[]
package example.micronaut

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Version

@MappedEntity // <1>
class Person(
    val name: String,
//end::clazz1[]
//tag::nullableage[]
    val age: Int?
//end::nullableage[]
/*
//tag::ageint[]
    val age: Int
//end::ageint[]
*/
//tag::clazz2[]
) {

    @Id // <2>
    @GeneratedValue // <3>
    var id: Long? = null

    @Version // <4>
    var version: Long? = null
}
//end::clazz2[]
